### DISCUSSION TITLE
Best practice for RWD with low-level styled component

### DISCUSSION THREAD

**Discussion by mattp0123** [stitchesjs/stitches#1050] [Help]
Low-level styled components are commonly seen in the [styled-system ecosystem](https://styled-system.com/ecosystem#ecosystem), like:
```typescript
const Flex = styled('div', {
  display: 'flex',
})
```
I can feel that it has good reusability and clear name for it's purpose. However, If the responsive breakpoint comes in, it will become really messy (IMO). like:
```typescript
<Flex
  css={{
    mt: 12,
    '@md': { mt: 15 },
    '@lg': { mt: 18 },
    '@xl': { mt: 21 },
  }}
>
  {content}
</Flex>
```
So I'm wondering what is the right situation for these components?

**Comment by thevipinmishra** [comment]
It can't get better (IMO), 

Right now, it's kind of best api for RWD solution on the consumption level.
